File: src/main/java/Main/SignalGenerator.java
package Main;

import org.bukkit.Bukkit;
import org.bukkit.plugin.Plugin;

public class SignalGenerator {
	private static SignalGenerator singleton = null;

	public static SignalGenerator inst()
	{
		if(singleton == null){
			singleton = new SignalGenerator();
		}
		return singleton;
	}

	private static Plugin plugin;
	public static Plugin getPlugin() {return plugin;}
	public void setPlugin(Plugin plugin) {this.plugin = plugin;}

	private static GameProcess gameProcess;
	private int TaskID = -1;

	//SignalGenerator() { }

	public void Initialize() { gameProcess = GameProcess.inst(); }

	//SignalGenerator() { gameProcess = GameProcess.inst(); }

	//===============================
	public void Cancel() {
		if(TaskID != -1)
			Bukkit.getScheduler().cancelTask(TaskID);
	}

	public void Start()
	{
		TaskID = Bukkit.getScheduler().scheduleSyncRepeatingTask(getPlugin(), new SignalGenerator20Hz(), 1L, 1L);
	}

	class SignalGenerator20Hz implements Runnable {

		@Override
		public void run() {
			gameProcess.CycleProgress();
		}
	}
}

File: src/main/kotlin/io/github/MEL_TEN/RandomSpawn/SignalGenerator.kt
package io.github.mel_ten.randomspawn

import org.bukkit.Bukkit
import org.bukkit.plugin.Plugin

class SignalGenerator {
    private val plugin = RandomSpawn.instance
    private var TaskID = -1

    //SignalGenerator() { }
    fun Initialize() {
        gameProcess = GameProcess.inst()
    }

    //SignalGenerator() { gameProcess = GameProcess.inst(); }
    //===============================
    fun Cancel() {
        if (TaskID != -1) Bukkit.getScheduler().cancelTask(TaskID)
    }

    fun Start() {
        TaskID = Bukkit.getScheduler().scheduleSyncRepeatingTask(plugin, SignalGenerator20Hz(), 1L, 1L)
    }

    internal inner class SignalGenerator20Hz : Runnable {
        override fun run() {
            gameProcess!!.CycleProgress()
        }
    }

    companion object {
        private var singleton: SignalGenerator? = null
        fun inst(): SignalGenerator? {
            if (singleton == null) {
                singleton = SignalGenerator()
            }
            return singleton
        }

        val plugin: Plugin? = null
        private var gameProcess: GameProcess? = null
    }
}